### PR TITLE
External Support Review

### PR DESCRIPTION
**Pull request by Amtresu**
##We are a team of 5 people, with an app that is located in two seperate repos.

Please use conor@gmail.com and password for an admin account. 

[Back end PR](https://github.com/PRATAP-KUMAR/book-an-appointment/pull/31)

- Apply technical knowledge and skills gained in previous modules in a complex project.
- Understand pros and cons of different approaches of connecting Ruby on Rails back-end with React front-end.
- Understand principles of Ruby on Rails and React frameworks.
- Apply Ruby best practices and language style guides in code.
- Apply RoR best practices and language style guides in code (e.g. thin controllers).
- Apply JavaScript best practices and language style guides in code.
- Apply React best practices and language style guides in code.
- Learn about and practice giving constructive feedback to teammates.
- Perform a code review for a team member.
- Use the "Review change"" feature from GitHub.
- Write clear, professional, and respectful review comments for other team members.
- Explain "why" a change is requested when giving a code review.
- Plan a 2+ week project with no interim Microverse-set milestones and submit it on time.
- Apply knowledge of setting working agreements to set group project teams up for success.
- Independently implement best practices for group agreements to improve teamwork in larger group project.
- Demonstrate ability to apply best practices of communication for resolving teamwork challenges.
- Understand that respect is the foundation of strong relationship-building with teammates.
- Show up throughout group projects as a reliable and committed team member who communicates and manages your time effectively.
- Recognize the importance of investing in good working relationships with teammates.
- Understand principles of strong teamwork (being reliable, committed, and consistent) and how to apply them in group projects.
- Recognize the value of making equal contributions to group projects to produce the best outcome.
- Use empty Kanban board to manage tasks with team and own time on the project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,46 +1,224 @@
...
+To run the project, execute the following command from the root directory:
+
+```sh
+  npm start
+```
+
+### Run tests
+````sh
+npm test
+````
```
Comment (Meltrust) [inline README.md]: - [ ] You have instructions to run tests, but your frontend app seems to have none.  Either add some tests or kindly remove this section. 👍

File: src/components/reservations/ReserveForm.js [hunk 2]
```diff
@@ -1,26 +1,66 @@
...
 /* eslint jsx-a11y/label-has-associated-control: ["error", { assert: "either" } ] */
-import React from 'react';
-import { Link } from 'react-router-dom';
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { Link, useNavigate } from 'react-router-dom';
+import { fetchCarsAsync } from '../../redux/cars/cars';
+import { reserveCarAsync } from '../../redux/reservations/reservations';
+
+function ReserveForm(props) {
```
Comment (Meltrust) [inline src/components/reservations/ReserveForm.js]: - [ ] When I create an account and go to "reservations", I can see I already have lots of them:


image

Making new reservations works, however, If I create another account I can see I have the same reservations, including the one I created on the last account.

This is not intended functionality, as every user should have their own reservations.

Kindly fix this bug so your app works as intended.  If the bug does not show up for you and your app needs special procedures to run it properly, kindly add them to your readme as instructions.

File: src/enroute.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+/* eslint-disable no-alert */
+const notification = 'You must be an admin to add car';
+
+const enroute = () => {
+  alert(notification);
```
Comment (Meltrust) [inline src/enroute.js]: - [ ] Your project has 2 alerts.  Here at Microverse, alerts are highly discouraged according to the javascript [requirements](https://github.com/microverseinc/curriculum-final-capstone/blob/main/projects/business_requirements.md#project-requirements):

image

You can use a modal, or a text that appears on the screen. 👍

File: src/redux/reservations/reservations.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+export const reserveCarAsync = (obj) => async (dispatch) => {
+  const objtwo = {
+    reserve_date: obj.reserve_date.slice(0, 10),
+    car_id: +obj.car_id,
+    user_id: obj.user_id,
+  };
+  const response = await axios.post(reserveCarAPI, objtwo);
+  const data = await response.data;
+  dispatch(reserveCar(data));
+  alert('data');
```
Comment (Meltrust) [inline src/redux/reservations/reservations.js]: - [ ] Your project has 2 alerts.  Here at Microverse, alerts are highly discouraged according to the javascript [requirements](https://github.com/microverseinc/curriculum-final-capstone/blob/main/projects/business_requirements.md#project-requirements):

image

You can use a modal, or a text that appears on the screen. 👍 

However, this one is a great candidate just to remove it, it only says "data".

File: src/components/auth/ProtectedRoutes.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import { Outlet } from 'react-router';
+import Login from './Login';
+
+const useAuth = (obj) => obj && obj?.authenticated;
+
+function ProtectedRoutes(props) {
+  const obj = props;
+  const { user } = obj;
+  const isAuth = useAuth(user);
+  console.log(isAuth);
```
Comment (Meltrust) [inline src/components/auth/ProtectedRoutes.js]: - [ ] According to the javascript [requirements](https://github.com/microverseinc/curriculum-final-capstone/blob/main/projects/business_requirements.md#project-requirements):

image

Therefore, kindly remove this console.log debugging code. 👍

File: src/App.js [hunk 6]
```diff
@@ -1,33 +1,61 @@
...
 function App() {
-  // const user = useSelector((state) => state);
-  // console.log(user);
+  // const navigate = useNavigate();
+
+  // const Enroute = () => {
+
+  //   alert(notification);
+  //   navigate('/cars');
+  // }
```
Comment (Meltrust) [inline src/App.js]: - [ ] Kindly, remove this old piece of commented code.

File: src/redux/users/users.js [hunk 7]
```diff
@@ -42,9 +45,12 @@ export const userLogIn = createAsyncThunk(SIGN_IN, async (obj, { dispatch }) =>
...
       type: SIGN_IN,
-      payload: data,
+      payload: {
+        ...data.data.data,
+        authenticated: true,
+      },
     });
-    localStorage.setItem('userDetails', JSON.stringify(data.data.data));
+    localStorage.setItem('userDetails', JSON.stringify({ ...data.data.data, authenticated: true }));
     // console.log('Data: ', data.data.data);
```
Comment (Meltrust) [inline src/redux/users/users.js]: - [ ] Kindly, remove this old piece of commented code and the other that is present in this file.

File: src/App.js [hunk 8]
```diff
@@ -1,33 +1,61 @@
...
+import ProtectedAdminRoutes from './components/auth/ProtectedAdminRoutes';
+import ProtectedRoutes from './components/auth/ProtectedRoutes';
 import './App.css';
+import './media-queries-breakpoints/Small.css';
+import './media-queries-breakpoints/Large.css';
+import './media-queries-breakpoints/XLarge.css';
+import './media-queries-breakpoints/XXLarge.css';
+import './media-queries-breakpoints/Desktop.css';
 
 function App() {
```
Comment (Meltrust) [inline src/App.js]: - [ ]  Please, kindly refactor this one and all other functions in your project to ES6 arrow functions, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered non-best practice.

You can do a search in Vscode for the "function" word, and correct as needed.  Shouldn't take you more than 5-10 min so don't worry:

image

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @Amtresu,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Beautiful frontend ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._

_I strongly recommend you take them into account as they can improve your final evaluation._





Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/components/reservations/ReserveForm.js (hunk 2 above)
Comment (itsmraga-hub) [reply src/components/reservations/ReserveForm.js]: I think our filter method had an issue, to resolve this we changed the url to be specific to user

File: README.md (hunk 1 above)
Comment (itsmraga-hub) [reply README.md]: Section removed

File: src/redux/reservations/reservations.js (hunk 4 above)
Comment (itsmraga-hub) [reply src/redux/reservations/reservations.js]: alert removed from here

File: src/App.js (hunk 8 above)
Comment (itsmraga-hub) [reply src/App.js]: All functions changed to ES6 arrow function

File: src/components/auth/ProtectedRoutes.js (hunk 5 above)
Comment (itsmraga-hub) [reply src/components/auth/ProtectedRoutes.js]: all console.logs removed

File: src/redux/users/users.js (hunk 7 above)
Comment (itsmraga-hub) [reply src/redux/users/users.js]: Removed. Thanks

File: src/App.js (hunk 6 above)
Comment (itsmraga-hub) [reply src/App.js]: Removed. Thanks

File: src/enroute.js (hunk 3 above)
Comment (itsmraga-hub) [reply src/enroute.js]: Changed the alert to a modal

**Review by Meltrust — APPROVED** [review]
#### Hi @Amtresu,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Test instructions have been fixed✔️ 
- "Shared" reservations bug has been fixed✔️ 
- Alerts have been removed, and modals have been created instead✔️
- Debugging code has been removed ✔️ 
- Commented code has been removed ✔️ 
- Almost all of the functions have been refactored to ES6 ✔️ 
- Excellent work, you've nailed this project ✔️ 

## Last suggestions:

- [x] OPTIONAL: Refactor remaining functions to ES6
- [x] OPTIONAL: After reserving a car and getting redirected to the "Reservations" page, I need to reload the page in order to get the new reservation to appear.  While trying to reproduce the issue, I noticed this does not happen always.  I recommend you take a look to see if you can reproduce it.

Cheers and Happy coding!👏👏👏


------

**Comment by itsmraga-hub** [comment]
Thanks @Meltrust, cheers and happy coding yourself. It's been a fun ride.

**Pull request merged**